Tests/SourceKitLSPTests/CodeActionTests.swift

```diff
@@ -482,4 +489,47 @@ final class CodeActionTests: XCTestCase {
     ]
     XCTAssertEqual(expectedCodeActions, codeActions)
   }
+
+  func testPackageManifestEditingCodeActionResult() async throws {
+    let testClient = try await TestSourceKitLSPClient(capabilities: clientCapabilitiesWithCodeActionSupport())
+    let uri = DocumentURI.for(.swift)
+    let positions = testClient.openDocument(
+      """
+      // swift-tools-version: 5.5
+      let package = Package(
+          name: "packages",
+          targets: [
+              .tar1️⃣get(name: "MyLib"),
+          ]
+      )
+      """,
+      uri: uri
+    )
+
+    let testPosition = positions["1️⃣"]
+    let request = CodeActionRequest(
+      range: Range(testPosition),
+      context: .init(),
+      textDocument: TextDocumentIdentifier(uri)
+    )
+    let result = try await testClient.send(request)
+
+    guard case .codeActions(let codeActions) = result else {
+      XCTFail("Expected code actions")
+      return
+    }
+
+    // Make sure we get the expected package manifest editing actions.
+    XCTAssertTrue(
+      codeActions.contains { action in
+        return action.title == "Add test target"
+      }
+    )
+
+    XCTAssertTrue(
+      codeActions.contains { action in
+        return action.title == "Add product to export this target"
+      }
+    )
+  }
```

Comment:
I think it would also be good to have tests for:
- Getting code actions on a `.testTarget` and ensure that we don’t get a code action to add a test target for it
- I think it makes sense to test for at least one code action that we get edits (so we test `asWorkspaceEdit`)
- I think we shouldn’t be giving these code actions in files that aren’t a package manifest, eg.
```swift
struct Foo {
  func target(name: String) {}
  func test() {
    self.target(name: "test")
  }
}
```

Reply:
SwiftPM does the checking to make sure we have a legitimate package file; we don't need to do it here.